rewrite-gradle/src/main/java/org/openrewrite/gradle/UpgradeDependencyVersion.java

```diff
@@ -416,6 +429,37 @@ private boolean shouldResolveVersion(String declaredGroupId, String declaredArti
                        new DependencyMatcher(groupId, artifactId, null).matches(declaredGroupId, declaredArtifactId);
             }
 
+            /**
+             * Extract property name from method invocation patterns like {@code property('guavaVersion')}
+             * or {@code findProperty('guavaVersion')}.
+             */
+            private @Nullable String extractPropertyNameFromMethodInvocation(J.MethodInvocation mi) {
+                String methodName = mi.getSimpleName();
+                if ("property".equals(methodName) || "findProperty".equals(methodName)) {
```

Comment:
I had an issue with using `MethodMatcher`. It seems that `MethodMatcher` doesn't work well with Groovy method invocations that lack full type attribution. Is there a better way to achieve this?

Reply:
Unfortunately no; @shanman190 has wrestled with this a lot as well. Tagging in case he has any new insights to share.

Reply:
Thanks! Pushed the change.